realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java

```diff
@@ -115,6 +115,87 @@ public void create_duplicatedNameThrows() {
     }
 
     @Test
+    public void createWithPrimaryKeyField_string() {
+        // Not nullable
+        realmSchema.createWithPrimaryKeyField("FooNonNull", "pkField", String.class, FieldAttribute.REQUIRED);
+        RealmObjectSchema objectSchema = realmSchema.getSchemaForClass("FooNonNull");
+        assertEquals("pkField", objectSchema.getPrimaryKey());
+        assertEquals(RealmFieldType.STRING, objectSchema.getFieldType("pkField"));
+        assertFalse(objectSchema.isNullable("pkField"));
+
+        // Nullable
+        realmSchema.createWithPrimaryKeyField("FooNull", "pkField", String.class);
+        objectSchema = realmSchema.getSchemaForClass("FooNull");
+        assertEquals("pkField", objectSchema.getPrimaryKey());
+        assertEquals(RealmFieldType.STRING, objectSchema.getFieldType("pkField"));
+        assertTrue(objectSchema.isNullable("pkField"));
+    }
+
+    @Test
+    public void createWithPrimaryKeyField_integer() {
+        // Not nullable
+        realmSchema.createWithPrimaryKeyField("FooNonNull", "pkField", int.class);
+        RealmObjectSchema objectSchema = realmSchema.getSchemaForClass("FooNonNull");
+        assertEquals("pkField", objectSchema.getPrimaryKey());
+        assertEquals(RealmFieldType.INTEGER, objectSchema.getFieldType("pkField"));
+        assertFalse(objectSchema.isNullable("pkField"));
+
+        // Nullable
+        realmSchema.createWithPrimaryKeyField("FooNull", "pkField", Integer.class);
+        objectSchema = realmSchema.getSchemaForClass("FooNull");
+        assertEquals("pkField", objectSchema.getPrimaryKey());
+        assertEquals(RealmFieldType.INTEGER, objectSchema.getFieldType("pkField"));
+        assertTrue(objectSchema.isNullable("pkField"));
+    }
+
+    @Test
+    public void createWithPrimaryKeyField_invalidClassNameThrows() {
+        String[] names = { null, "", TestHelper.getRandomString(57) };
```

Comment:
Shouldn't we avoid to have U+0000 or any other chars not valid for table name in the name?
I think we should update `TestHelper.getRandomString(int)` to exclude U+0000 at least.